### PR TITLE
Log statement and progress

### PR DESCRIPTION
**Pull request by jovi-s**
Addition of logging mechanisms in the command line for input modes:

1. Webcam: Log every n number of frames passed
2. Video: Logs approximate completion of video in steps of 10%

Resolves Issue #253 and #254

### CODE REVIEW

File: peekingduck/pipeline/nodes/input/recorded.py [hunk 1]
```diff
@@ -53,9 +55,19 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         '''
         outputs = self._run_single_file()
 
+        _processing = round((self.frame_counter/self.videocap.frame_count)*100)
+        self.frame_counter += 1
+
+        if _processing > self.tens_count:
+            self.logger.info('Approximate Processed: %s ...', self.tens_count)
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/recorded.py:62]: - Include percentage for clarity
- Approximate**ly**
```
Approximately processed: 10% ...
```

File: peekingduck/pipeline/nodes/input/utils/read.py [hunk 2]
```diff
@@ -118,3 +118,13 @@ def resolution(self) -> Tuple[int, int]:
         width = self.stream.get(cv2.CAP_PROP_FRAME_WIDTH)
         height = self.stream.get(cv2.CAP_PROP_FRAME_HEIGHT)
         return int(width), int(height)
+
+    @property
+    def frame_count(self) -> float:
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/utils/read.py:123]: It should return an "int" type not "float"?

File: peekingduck/pipeline/nodes/input/recorded.py [hunk 3]
```diff
@@ -33,6 +33,8 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self.resize_info = config['resize']
         self._mirror_image = config['mirror_image']
         self.file_end = False
+        self.frame_counter = 0
+        self.tens_count = 10
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/recorded.py:37]: For consistency, suggest either both of them to be `counter` or `count`

File: peekingduck/pipeline/nodes/input/recorded.py [hunk 4]
```diff
@@ -53,9 +55,19 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         '''
         outputs = self._run_single_file()
 
+        _processing = round((self.frame_counter/self.videocap.frame_count)*100)
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/recorded.py:58]: We usually have an underscore in front for private functions, but not for variables.

File: peekingduck/pipeline/nodes/input/live.py [hunk 5]
```diff
@@ -56,6 +58,10 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
                         "pipeline_end": False,
                         "filename": self.filename,
                         "fps": self.fps_saved_output_video}
+            self.frame_counter += 1
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/live.py:61]: Here's an interesting point to ponder about - will integer overflow be a problem?
https://stackoverflow.com/questions/52151647/integer-overflow-in-python3

If not, will `self.frame_counter` become so big that it eats up a lot of memory?
https://code.tutsplus.com/tutorials/understand-how-much-memory-your-python-objects-use--cms-25609

File: peekingduck/pipeline/nodes/input/utils/read.py [hunk 6]
```diff
@@ -118,3 +118,13 @@ def resolution(self) -> Tuple[int, int]:
...
+        """ Get total number of frames of file
+
+        Returns:
+            int: number indicating frame count
+        """
+        num_frames = self.stream.get(cv2.CAP_PROP_FRAME_COUNT)
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/utils/read.py:129]: It appears to work fine on images, but out of curiosity, what is the value of `cv2.CAP_PROP_FRAME_COUNT` on a single image? Does it return 1?

File: peekingduck/configs/input/live.yml [hunk 7]
```diff
@@ -9,4 +9,5 @@ resize: {
             height: 720
         }
 input_source: 0    # 0 for webcam, rtsp URL for CCTV
-mirror_image: False
\ No newline at end of file
+mirror_image: False
+frames_log_freq: 100    # Showcases every number of frames passed
```
Comment (leeping-ng) [inline peekingduck/configs/input/live.yml:13]: Can the comment be made clearer?

File: peekingduck/pipeline/nodes/input/utils/read.py (hunk 6 above)
Comment (jovi-s) [reply peekingduck/pipeline/nodes/input/utils/read.py:129]: yes, I believe so.

**Pull request merged**